versioned/storage/common-tests/src/main/java/org/projectnessie/versioned/storage/commontests/AbstractIndexesLogicTests.java

```diff
@@ -338,4 +342,79 @@ public void perCommitOps() {
 
     soft.assertThat(indexesLogic.commitOperations(commit)).containsExactly(add1, add2, remove1);
   }
+
+  @SuppressWarnings("unused")
+  static StoreConfig.Adjustable persistWithSmallIndexSize(StoreConfig.Adjustable config) {
+    return config.withMaxIncrementalIndexSize(200);
+  }
+
+  @Test
+  void completeIndexesInCommitChainWithStripes(
+      @NessiePersist(configMethod = "persistWithSmallIndexSize") Persist persist) throws Exception {
+
+    ObjId currentId = fiveIncompleteCommitsWithThreeOpsEach(persist);
+
+    IndexesLogic indexesLogic = indexesLogic(persist);
+    indexesLogic.completeIndexesInCommitChain(currentId, () -> {});
+
+    for (int i = 5; i > 0; i--) {
+      CommitObj current = persist.fetchTypedObj(currentId, ObjType.COMMIT, CommitObj.class);
+      if (i > 1) {
+        // All commits except commit 1 should have stripes
+        soft.assertThat(current.referenceIndexStripes()).isNotEmpty();
+      }
+      int actualElementCount =
+          indexesLogic.incrementalIndexFromCommit(current).elementCount()
+              + (current.hasReferenceIndex()
+                  ? requireNonNull(indexesLogic.buildReferenceIndexOnly(current)).elementCount()
+                  : 0);
+      int expectedElementCount = i * 3;
+      soft.assertThat(actualElementCount).isEqualTo(expectedElementCount);
+      currentId = current.directParent();
+    }
+  }
+
+  private ObjId fiveIncompleteCommitsWithThreeOpsEach(Persist persist) throws ObjTooLargeException {
+
+    String[][] keys =
+        new String[][] {
+          {"abcd", "bcde", "cdef"},
+          {"efgh", "fghi", "ghij"},
+          {"hijk", "ijkl", "jklm"},
+          {"klmn", "lmno", "mnop"},
+          {"nopq", "opqr", "pqrs"},
+        };
+
+    CommitObj current = null;
+    for (int i = 0; i < 5; i++) {
+
+      ObjId parentId = current == null ? EMPTY_OBJ_ID : current.id();
+
+      CommitObj.Builder c =
+          commitBuilder()
+              .id(randomObjId())
+              .addTail(parentId)
+              .created(System.nanoTime())
+              .seq(i)
+              .message("Commit " + (i + 1))
+              .headers(EMPTY_COMMIT_HEADERS)
+              .incompleteIndex(true);
+
+      StoreIndex<CommitOp> index = newStoreIndex(CommitOp.COMMIT_OP_SERIALIZER);
+
+      for (int j = 0; j < 3; j++) {
+        StoreKey k = key(keys[i][j]);
+        UUID cid = new UUID(123L, j);
+        ObjId v = randomObjId();
+        index.add(indexElement(k, commitOp(ADD, 42, v, cid)));
+      }
+
+      // incremental index has size 175, max configured is 200
```

Comment:
Can you make that `200` a constant and add an assertion here, like this (I know, a bit paranoid):
```java
soft.assertThat(serializedSize * 4 / 3 > THAT_CONSTANT);
```